addons/console_and_textchat/console.gd

```diff
@@ -2,268 +2,389 @@
 
 
 func _ready():
-	_build_console()
-	set("minimum_size", size)
-	_register_premade_commands()
-
-	resized.connect(_on_text_window_resized)
-
+    _build_console()
+    set("minimum_size", size)
+    _register_premade_commands()
+    _handle_cli_themes()
+    _handle_cli_visibility()
+    resized.connect(_on_text_window_resized)
 
 
 func _build_console() -> void:
-	_format_panel()
-	_instance_and_setup_children()
+    _format_panel()
+    _instance_and_setup_children()
 
 
 func _format_panel() -> void:
-	#custom_minimum_size = Vector2(300,335)
-	size_flags_horizontal = Control.SIZE_EXPAND_FILL
-	size_flags_vertical = Control.SIZE_EXPAND_FILL
+    #custom_minimum_size = Vector2(300,335)
+    size_flags_horizontal = Control.SIZE_EXPAND_FILL
+    size_flags_vertical = Control.SIZE_EXPAND_FILL
 
 
 func _instance_and_setup_children() -> void:
-	var vbox: VBoxContainer = VBoxContainer.new()
-	add_child(vbox)
+    var vbox: VBoxContainer = VBoxContainer.new()
+    add_child(vbox)
 
-	_instance_and_setup_chat_window(vbox)
-	_instance_and_setup_text_input(vbox)
+    _instance_and_setup_chat_window(vbox)
+    _instance_and_setup_text_input(vbox)
 
 
 func _instance_and_setup_chat_window(parent: VBoxContainer) -> void:
-	var display: RichTextLabel = RichTextLabel.new()
-	display.name = "DisplayLabel"
-	#display.custom_minimum_size = Vector2(300,300)
-	display.bbcode_enabled = true
-	display.scroll_active = true
-	display.scroll_following = true
-	display.clip_contents = false
-	display.size_flags_vertical = Control.SIZE_EXPAND_FILL
-	parent.add_child(display)
-	
-	display_label = display
+    var display: RichTextLabel = RichTextLabel.new()
+    display.name = "DisplayLabel"
+    #display.custom_minimum_size = Vector2(300,300)
+    display.bbcode_enabled = true
+    display.scroll_active = true
+    display.scroll_following = true
+    display.clip_contents = false
+    display.size_flags_vertical = Control.SIZE_EXPAND_FILL
+    parent.add_child(display)
+
+    display_label = display
 
 
 func _instance_and_setup_text_input(parent: VBoxContainer) -> void:
-	var edit: LineEdit = LineEdit.new()
-	edit.name = "TextInputLine"
-	edit.custom_minimum_size = Vector2(300,35)
-	edit.placeholder_text = "Insert message or command"
-	edit.text_submitted.connect(_on_text_input_line_text_submitted.bind())
+    var edit: LineEdit = LineEdit.new()
+    edit.name = "TextInputLine"
+    edit.custom_minimum_size = Vector2(300, 35)
+    edit.placeholder_text = "Insert message or command"
+    edit.text_submitted.connect(_on_text_input_line_text_submitted.bind())
 
-	parent.add_child(edit)
-	text_input_line = edit
+    parent.add_child(edit)
+    text_input_line = edit
 
 
 ####################################################################################################
 ## CONSOLE OUTPUT ##
 
-func print_message(text: String) -> void:
-	display_label.text += "\n"
-	display_label.text += text
+func print_message(text: String, codeblock_replacement = true) -> void:
```

Comment:
I must admit, that I do not understand the purpose of changes in that function at all.

Reply:
I think this was a leftover from wanting to support some markdown-styled code blocks or something. I'm going to remove the changes to this function since it's half-baked and not used anywhere.